Create launchpad for launching futarchy-native DAOs

metaproph3t opened the pull request.

metaproph3t committed:
Improve `initialize_launch` test

metaproph3t committed:
Create `launchpad` with `initialize_launch`

metaproph3t committed:
Update SDK version

metaproph3t committed:
Test `complete_launch`

metaproph3t committed:
Fix `initialize_launch` and `fund` for uncapped supply

metaproph3t committed:
Add CPI event

metaproph3t committed:
Add `complete_launch`

metaproph3t committed:
Add refund

metaproph3t committed:
Add Raydium program

metaproph3t committed:
Use 3 instead of 10 SOL for launch

metaproph3t committed:
Update SDK version

metaproph3t committed:
Complete launchpad 🚀

metaproph3t committed:
v0.4.0-alpha.37

metaproph3t committed:
Fix test

metaproph3t committed:
v0.4.0-alpha.38

metaproph3t committed:
Fix `token_0` and `token_1`

metaproph3t committed:
Start integrating Raydium

metaproph3t committed:
Get Raydium deposit working

metaproph3t committed:
Add CPI events

metaproph3t committed:
Add `start_launch`

metaproph3t committed:
Add timer

metaproph3t committed:
Update yarn version

metaproph3t committed:
Make it 1,000 tokens per USDC instead of 10,000

staccDOTsol commented:
hi

metaproph3t commented:
> hi

henlo

staccDOTsol commented:
lob u. water u doing here. how can I propheet prophet

staccDOTsol commented:
I actually just wanted email updates on pushes (but comments are fun)

metaproph3t commented:
i lob u long tim

staccDOTsol commented:
we gonna push this hard ser? wen wen wen

metaproph3t commented:
soon

staccDOTsol commented:
bold of you to not allow ppl to defund unless not raised
is ui being built alongside or is it an afterthing
is there a timeline on this thing of beauty

one q I have is will the ui now display all futarchies regardless origin or only metadao+launchpads

elrakabawi commented:
Supp guys? I'm a frequent visitor of this PR as well. Can offer help on UI if needed.

staccDOTsol commented:
good progress here meta, bigups

metaproph3t committed:
Switch `launch` signer to `launch_signer` for mint authority

metaproph3t committed:
Update version for tests

metaproph3t committed:
Move mint authority to dao treasury post-launch

metaproph3t committed:
v0.4.0-alpha.41

metaproph3t committed:
v0.4.0-alpha.39

metaproph3t committed:
Update version

metaproph3t committed:
Rename `launch_treasury` -> `launch_signer`

Less confusing because you have the DAO treasury

metaproph3t committed:
Add program ID to README

metaproph3t committed:
Update SDK version for tests

metaproph3t committed:
Add freeze authority check to `ininitialize_launch`

metaproph3t committed:
Add `seq_num` to launchpad CPI events

metaproph3t committed:
v0.4.0-alpha.40

metaproph3t committed:
Add `initializeLaunch` script

metaproph3t committed:
Add events to `autocrat`

metaproph3t committed:
Add GitHub action to generate verifiable launchpad

metaproph3t committed:
Remove deprecated scripts

github-actions[bot] committed:
Update amm verifiable build

github-actions[bot] committed:
Update conditional_vault verifiable build

github-actions[bot] committed:
Update launchpad verifiable build

github-actions[bot] committed:
Update autocrat verifiable build

metaproph3t committed:
Add the solana program extend command

metaproph3t committed:
Add a script to show program size diffs

metaproph3t committed:
Change `initialize_launch` to use `emit_cpi` over `emit`

metaproph3t committed:
Fix the Raydium AMM config

metaproph3t committed:
Some cleanup

metaproph3t committed:
Modify the system to use 10m tokens

metaproph3t committed:
Fix tests for new SDK

metaproph3t committed:
Move to dynamic length launches

metaproph3t committed:
Add `launch` to `FundingRecord`

metaproph3t committed:
Update the launch script

metaproph3t committed:
Switch `fund` to use `funding_record`

People are concerned about users creating liquidity pools while a raise
is active, so I'm switching to a claim-based system.

metaproph3t committed:
v0.4.0-alpha.42

metaproph3t committed:
Add extra token mint check

metaproph3t committed:
Get launchpad working on devnet

metaproph3t committed:
Update SDK version

metaproph3t committed:
Fix bug with launch signer PDA seeds

metaproph3t committed:
Flip the quote and base for `initialize_dao`

metaproph3t committed:
Final changes for pre-audit

Make it so that DAOs are initialized in the completion state, which we
need to do then because we don't know the price otherwise

metaproph3t committed:
Add create proposal script

metaproph3t committed:
Add funding record to refund event

metaproph3t committed:
Add MTN

metaproph3t committed:
Add metadata to `initialize_launch`

pileks committed:
update launchpad events for simpler historic indexing

metaproph3t committed:
Clean up README

metaproph3t committed:
Update SDK versions

metaproph3t committed:
Merge pull request #265 from metaDAOproject/pileks/launchpad-event-updates

Update launchpad events for simpler historic indexing

pileks committed:
bump package sdk version

metaproph3t committed:
Verify that launch isn't expired in fund

metaproph3t committed:
Use normal USDC and check that it's USDC

metaproph3t committed:
Replace slots with unix timestamp

metaproph3t committed:
Make launch DAOs PDAs to prevent griefing

Otherwise, someone could grind an offensive-name DAO

metaproph3t committed:
Cargo fmt

metaproph3t committed:
Adopt the payer-signer pattern

pileks committed:
add dao and dao_treasury to LaunchCompletedEvent

pileks committed:
Merge branch 'launchpad' of github.com:metaDAOproject/futarchy into launchpad

metaproph3t committed:
v0.4.0-alpha.49

metaproph3t committed:
Conditionally compile the minimum bound on launch seconds

metaproph3t committed:
Update SDK version from main

metaproph3t committed:
v0.4.0-alpha.48

metaproph3t committed:
Get USDC working on devnet

metaproph3t committed:
Update refund for devnet USDC

pileks committed:
bump sdk package version, release new sdk

metaproph3t committed:
Implement payer-signer pattern for `fund`

metaproph3t committed:
Make it so that `complete_launch` sends the LP tokens

metaproph3t committed:
Transfer token metadata authority to new DAO

metaproph3t committed:
Remove unnecessary `msg!`

metaproph3t committed:
Add check that last 4 chars of token are 'meta'

metaproph3t committed:
Add audits folder

metaproph3t committed:
Update dep in main

metaproph3t committed:
Cargo fmt

metaproph3t committed:
v0.4.0-alpha.51

metaproph3t committed:
Update to latest SDK

metaproph3t committed:
v0.4.0-alpha.50

metaproph3t committed:
Add full launch integration test

metaproph3t committed:
Yarn lint

suite committed:
added v0 dao and proposal scripts

suite committed:
Merge branch 'launchpad' of github.com:metaDAOproject/futarchy into launchpad
